redis/asyncio/connection.py

```diff
@@ -787,6 +791,7 @@ async def read_response(
         self,
         disable_decoding: bool = False,
         timeout: Optional[float] = None,
+        disconnect_on_error: bool = True,
```

Comment:
This function starts having a lot of position arguments that are of primitive types. Consider making additional ones kw-only (we can't change existing ones without making a breaking change):
```suggestion
        *,
        disconnect_on_error: bool = True,
```

Reply:
Good suggestion.  I'll add that to the `timeout` too, since that's one that I added and it is used only as a kwarg.